### ISSUE TITLE
Data bag integration with user cookbook

### ISSUE THREAD

**Issue by Joseph-R** [hw-cookbooks/chef-server-populator#40]
Hey guys,

I'm building a PoC chef server and attempting to deploy the whole thing from code using this cookbook, chef-server, etc.   

It's not clear to me if there is an intended integration or interoperability between the data bags used in this cookbook for user/client registration (`node[:chef_server_populator][:databag]) and the user data bags from Chef's [users](https://github.com/opscode-cookbooks/users)?

I'd prefer not to create another data bag with duplicate users and have to maintain code in two places if I can help it.  Is there any integration available/intended with this or other chef user management tools, at this time?  

I'm assuming not.  But it would be really handy if I could use the same data bag for each and have this cookbook only register clients, keys, etc, if a certain value existed in each user's file.  

Example user entry in the format used by the Chef-maintained user cookbook:
```
{
  "id": "user_id",
  "ssh_keys": [
    "key_value"
    ],
  "groups": [ "developers" ],
  "uid": 607,
  "shell": "\/bin\/bash",
  "comment": "comment_valuel"
}
```

Which I'm thinking could become:
```
{
    "id": "user_id",
    "full_name": "User Name",
    "ssh_keys": [
        "key_value"
    ],
    "groups": [
        "developers"
    ],
    "uid": 607,
    "email": "name@domain.tld",
    "shell": "/bin/bash",
    "comment": "comment_value",
    "chef_server": {
        "client_key": "public key contents",
        "type": [
            "user"
        ],
        "orgs": {
            "organization": {
                "enabled": true,
                "admin": true
            }
        }
    }
}
```

**Comment by luckymike** [comment]
That is in fact exactly the intention. If you look at the client recipe, you'll see that we're checking for the `chef_server` hash in the data bag item, and ignoring items without it.

**Comment by Joseph-R** [comment]
@luckymike - Gotcha!  Great, thanks!  I appreciate the speedy feedback.

So this seems like a great cookbook and it *looks* like it does what want it to do (ie, take me from totally fresh chef-server up to a configured or restored chef-server).  

One suggestion, coming at this with completely fresh eyes, is that the documentation for $brand_spanking_new_guy could be a little clearer.

1.  The two usage scenarios are chef-solo and chef-client.  It's not clear where chef-zero falls.  
2.  Assuming (based on the kitchen file) that I could run chef-zero in kitchen and either a) treat this like a client run or b) just call chef-server-populator and let the code figure it out...
3. I ran into [this error](https://gist.github.com/Joseph-R/27454da118db7327a78f) running the following code:

Databag entry for organization, where node['chef_server_populator']['databag'] = "users":
```
{
  "id": "jdpa_o2o",
  "full_name": "JD Power and Associates - O2O",
  "chef_server": {
    "client_key": "key_value",
    "type": [
      "org"
    ],
    "enabled": true
  }
}
```

Databag entry for user: 
```
{
    "id": "jreid",
    "full_name": "joseph reid",
    "ssh_keys": [
        "key_value"
    ],
    "groups": [ "sysadmin", "developers" ],
    "uid": 602,
    "email": "joe.reid@email.com",
    "shell": "\/bin\/bash",
    "comment": "Joe Reid",
    "chef_server": {
        "client_key": "key_value",
        "type": [ "user" ],
        "orgs": {
            "organization": {
                "enabled": true,
                "admin": true
            }
        }
    }
}
```

And this (snippet) of my kitchen.yml
```
suites:
  - name: server
    driver:
      driver:
      - ["private_network", {ip: "33.33.33.209"}]
      customize:
        memory: 2048
    run_list:
      - recipe[O2O-chefserver::default]
    attributes:
      chef_server_populator:
        endpoint: localhost
        databag: users
```

Where the default recipe is simply:
```
include_recipe "chef-server::default"
include_recipe "chef-server-populator"
```

I solved that error by changing the code in my user.json from "organization" to "$organization_name":
```
        "orgs": {
            "jdpa_o2o": {
                "enabled": true,
                "admin": true
            }
        }
```
...which is fine, but was not immediately obvious to me.

If you guys are open to it, I can submit a PR with some "setup for dummies" clarification steps in there.  

Regards,
Joe Reid

**Comment by luckymike** [comment]
We :heart: PRs! You're right, using `organization` as a pseudo org name is confusing.

**Comment by Joseph-R** [comment]
Thanks, man!  Will do.

I've got another question about setting up the validation client and key.  (I can break these off and post them separately, if need be.  Just let me know.)

At this point I am trying to configure my knife.rb to make sure it connect to the local chef server (kitchen VM).  I am not seeing a validator key under /etc/chef on the VM, and I am not sure if I have configured one, if those are managed by this cookbook, and generally "how to deal with the validator key?"

I have databag items for creating a default organization and one user (myself), assigned to that organization, with admin rights.  I am not seeing a validator key under /etc/chef.  Is that something handled by the cookbook, or am I missing a step?

**org.json:**
```
{
  "id": "jdpa_o2o",
  "full_name": "JD Power and Associates - O2O",
  "chef_server": {
    "client_key": "key_value",
    "type": [
      "org"
    ],
    "enabled": true
  }
}
```

**Output from on the test VM:**
```
[root@server-centos-65 chef]# chef-server-ctl org-show
jdpa_o2o: https://127.0.0.1/organizations/jdpa_o2o

[root@server-centos-65 chef]# chef-server-ctl user-list
jreid
pivotal
```

Regards,
Joe Reid
